backend/main.go

```diff
@@ -42,20 +45,61 @@ const (
 type Server struct {
 	store        storage.StorageService
 	storeTimeout time.Duration
+	auth         Auth
+}
+
+type Auth interface {
+	authenticate(next http.HandlerFunc) http.HandlerFunc
+}
+
+type noAuth struct{}
+
+func (auth noAuth) authenticate(next http.HandlerFunc) http.HandlerFunc {
+	return next
+}
+
+type firebaseAuth struct {
+	firebaseClient authClient
+}
+
+type authClient interface {
```

Comment:
I believe this is the interface you're writing specifically to match our expectations of the Firebase client, so we can mock it out in tests, correct?

If so then it should probably have "Firebase" somewhere in its name. If you're moving these types to another package, we should probably have a file in there called `firebase.go` which contains all the Firebase-specific code.

Reply:
Renamed to firebaseAuthClient